### ISSUE TITLE
Failure to converge when linked with Intel MKL

### ISSUE THREAD

**Issue by rmalouf** [fchollet/keras#996]
I've run into a problem that I'm not sure where to even begin debugging.  I'm getting wildly different results when I run the same program on different hardware.  It has something to do with the Intel MKL BLAS -- when I use a version of numpy linked with MKL, the results are terrible. To take one example, when I run a particular network using MKL BLAS and device=cpu, the loss starts at around 10 and drifts a but up and down but never consistently decreases .  With MKL BLAS and device=gpu, the loss drops quick to 5 and then hovers around that.  Using numpy linked with OpenBLAS, the loss steadily decreases from 10 to (after 1500 epochs) about 0.003.  

Has anyone reported something like this before?  It's not a critical bug for me, since the solution is obvious: just use OpenBLAS.  But the community ought to be concerned about this.  As I say, I don't know where to begin to isolate this... whether it's a problem with MKL, numpy, theano, keras, the compilers, or some combination of all of the above.

**Comment by the-moliver** [comment]
If I had to guess I'd say the issue is probably in theano. Can you post the minimal code necessary to reproduce?

**Comment by rmalouf** [comment]
Well, it's not minimal but the problem shows up with the addition_rnn.py example, but not reuters_mlp.py.

But: I don't know why I didn't think to do this first, but I ran the full numpy test suite and it turned up some very alarming looking errors.  I'll follow those up with the numpy folks, and hopefully fix that will fix the problems with keras.  Sorry for the false alarm!
